foxglove/console/api.go

```diff
@@ -471,6 +471,68 @@ func (e ExtensionResponse) String() string {
 	return fmt.Sprintf("%s.%s-%s", e.Publisher, e.Name, *version)
 }
 
+type PendingImportsRequest struct {
+	RequestId       string `json:"requestId" form:"requestId,omitempty"`
+	DeviceId        string `json:"device.id" form:"device.id,omitempty"`
+	DeviceName      string `json:"device.name" form:"device.name,omitempty"`
+	Error           string `json:"error" form:"error,omitempty"`
+	Filename        string `json:"filename" form:"filename,omitempty"`
+	ShowCompleted   bool   `json:"showCompleted" form:"showCompleted,omitempty"`
+	ShowQuarantined bool   `json:"showQuarantined" form:"showQuarantined,omitempty"`
+	SiteId          string `json:"siteId" form:"siteId,omitempty"`
+}
+
+type PendingImportsResponseItem struct {
+	CreatedAt     time.Time `json:"createdAt"`
+	UpdatedAt     time.Time `json:"updatedAt"`
+	OrgId         *string   `json:"orgId"`
+	Filename      *string   `json:"filename"`
+	PipelineStage *string   `json:"pipelineStage"`
+	RequestId     *string   `json:"requestId"`
+	DeviceId      string    `json:"deviceId"`
+	DeviceName    string    `json:"deviceName"`
+	ImportId      string    `json:"importId"`
+	SiteId        string    `json:"siteId"`
+	Status        string    `json:"status"`
+	Error         string    `json:"error"`
+}
+
+func (r PendingImportsResponseItem) Fields() []string {
+	createdAt := r.CreatedAt.Format(time.RFC3339)
+	updatedAt := r.UpdatedAt.Format(time.RFC3339)
+	return []string{
+		requiredVal(&createdAt),
+		requiredVal(&updatedAt),
+		requiredVal(r.OrgId),
+		requiredVal(r.Filename),
+		requiredVal(r.PipelineStage),
+		requiredVal(r.RequestId),
```

Comment:
I think we can drop these `requiredVal` calls. There's probably a better name for this function, but it's ensuring that we output either the value or an empty string for those fields. There's no need to call that when the API guarantees them in the response.

Reply:
if they are guaranteed in the response, we can avoid typing them as pointers